src/Symfony/Component/Serializer/Mapping/Loader/YamlFileLoader.php

```diff
@@ -75,9 +76,13 @@ public function loadClassMetadata(ClassMetadataInterface $classMetadata)
 
                     if (isset($data['groups'])) {
                         foreach ($data['groups'] as $group) {
-                            $attributeMetadata->addGroup($group);
+                            $attributeMetadata->addGroup((string) $group);
                         }
                     }
+
+                    if (isset($data['max_depth'])) {
+                        $attributeMetadata->setMaxDepth((int) $data['max_depth']);
```

Comment:
We should check the value (we do that for the annotation and we have an implicit check in the XML loader due to the XSD file).